src/Build/Construction/Solution/SolutionFile.cs

```diff
@@ -195,7 +197,16 @@ internal string FullPath
             {
                 // Should already be canonicalized to a full path
                 ErrorUtilities.VerifyThrowInternalRooted(value);
-                _solutionFile = value;
+                if (FileUtilities.IsSolutionFilterFilename(value))
+                {
+                    _solutionFilterFile = value;
+                    _solutionFile = value.Substring(0, value.Length - 1);
```

Comment:
Don't you need to parse the solution filename out of the slnf? For example, in Roslyn's `Compilers.slnf` it points to `Roslyn.sln`:

https://github.com/dotnet/roslyn/blob/7bc44488c661fd6bbb6c53f39512a6fe0cc5ef84/Compilers.slnf#L3